docs/en_US/Compression/Pruner.rst

```diff
@@ -722,3 +723,128 @@ User configuration for Sensitivity Pruner
+       2. Sort heads locally or globally, and prune out some heads with lowest scores. The number of pruned heads is determined according to the sparsity specified in the config.
+       3. If the specified pruning iteration is larger than 1 (iterative pruning), finetune the model for a while before the next pruning iteration.
+
+Currently, the following head sorting criteria are supported:
+
+    * "l1_weight": rank heads by the L1-norm of weights of the query, key, and value projection matrices.
+    * "l2_weight": rank heads by the L2-norm of weights of the query, key, and value projection matrices.
+    * "l1_activation": rank heads by the L1-norm of their attention computation output.
+    * "l2_activation": rank heads by the L2-norm of their attention computation output.
+    * "taylorfo": rank heads by l1 norm of the output of attention computation * gradient for this output. Check more details in `this paper <https://arxiv.org/abs/1905.10650>`__ and `this one <https://arxiv.org/abs/1611.06440>`__.
+
+We support local sorting (i.e., sorting heads within a layer) and global sorting (sorting all heads together), and you can control by setting the ``global_sort`` parameter. Note that if ``global_sort=True`` is passed, all weights must have the same sparsity in the config list. However, this does not mean that each layer will be prune to the same sparsity as specified. This sparsity value will be interpreted as a global sparsity, and each layer is likely to have different sparsity after pruning by global sort.
+
+In our implementation, we support two ways to group the four weights in the same layer together. You can either pass a nested list containing the names of these modules (usage 1 below) to the pruner, or simply pass a dummy input and the pruner will run ``torch.jit.trace`` to group the weights (usage 2 below).
+
+However, if you would like to assign different sparsity to each layer, currently you could only use the first option, i.e., passing names of the weights to the pruner (usage 3 below). Also note that weights belong to the same layer must have the same sparsity.
+
+In addition to the following usage guide, we provide a more detailed example of pruning BERT for tasks from the GLUE benchmark. Please find it in this :githublink:`page <examples/model_compress/pruning/transformers>`.
+
+Usage
+^^^^^
+
+Usage 1: one-shot pruning, same sparsity for all the layers (PyTorch code)
+
+.. code-block:: python
+
+   from nni.algorithms.compression.pytorch.pruning import TransformerHeadPruner
+   kwargs = {'ranking_criterion': "l1_weight",
+             'global_sort': False,
+             'num_iterations': 1,
+             'epochs_per_iteration': 1,        # this is ignored when num_iterations = 1
+             'head_hidden_dim': 64,
+             'dummy_input': dummy_input,
+             'trainer': trainer,
+             'optimizer': optimizer
+             }
+    config_list = [{
+        'sparsity': 0.5,
+        'op_types': ["Linear"]
```

Comment:
Why Linear here? feel a little bit confused

Reply:
Currently we can only handle attention layers with their projection weights written as separate Linear modules. I will further clarify this in the docs and the example.

Reply:
improved docs

Reply:
Yes, I mean if this config list is passed to TransformerHeadPruner(which is only used to prune the head only), then why should we need the `op_type` attribute, not to mention `Linear` is not the type of attention head.

Reply:
I think `Compressor`'s config validation requires that either `op_types` or `op_names` is specified. But more importantly, if the user does not give the `attention_name_groups` parameter, then the user is also unlikely to give `op_names`. As a result, we will obtain graph and infer the name groups, and this process basically requires the `op_type` being `Linear`. So I think by letting the user write out explicitly we can remind them that we can only handle the scenario where different projection weights are represented as separate `Linear` layers.

Reply:
does usage 1 reliable?

Reply:
i suggest we only show usage 3 in doc

Reply:
> does usage 1 reliable?

It is reliable for Huggingface transformers, but may not work for other implementations. In that case, the user would have to try usage 2.

Reply:
> i suggest we only show usage 3 in doc

I will first promote usage 3 to the front, and delete usage 2

Reply:
fixed docs